mmtrack/models/sot/stark.py

```diff
@@ -0,0 +1,345 @@
+                      output_size):
+        """ Crop Image
+        Only used during testing
+        This function mainly contains two steps:
+        1. Crop `img` based on target_bbox and search_area_factor. If the
+        cropped image/mask is out of boundary of `img`, use 0 to pad.
+        2. Resize the cropped image/mask to `output_size`.
+
+        args:
+            img (Tensor): of shape (B, C, H, W)
+            target_bbox (list | ndarray | array): in [cx, cy, w, h] format
+            search_area_factor (float): Ratio of crop size to target size
+            output_size (float): Size to which the extracted crop is resized (always square).
+        returns:
+            img_crop_padded (Tensor): of shape (B, C, H, W)
+            resize_factor (float): the factor by which the crop has been resized to make the
+                crop size equal output_size
+            att_mask (ndarray)
+        """
+        cx, cy, w, h = target_bbox.split((1, 1, 1, 1), dim=-1)
+
+        _, _, img_h, img_w = img.shape
+        # 1. Crop image
+        # 1.1 calculate crop size and pad size
+        crop_size = math.ceil(math.sqrt(w * h) * search_area_factor)
+        if crop_size < 1:
+            raise Exception('Too small bounding box.')
+
+        x1 = torch.round(cx - crop_size * 0.5).long()
+        x2 = x1 + crop_size
+        y1 = torch.round(cy - crop_size * 0.5).long()
+        y2 = y1 + crop_size
+
+        x1_pad = max(0, -x1)
+        x2_pad = max(x2 - img_w + 1, 0)
+        y1_pad = max(0, -y1)
+        y2_pad = max(y2 - img_h + 1, 0)
+
+        # 1.2 crop image
+        img_crop = img[..., y1 + y1_pad:y2 - y2_pad, x1 + x1_pad:x2 - x2_pad]
+
+        # 1.3 pad image
+        img_crop_padded = F.pad(
+            img_crop,
+            pad=(x1_pad, x2_pad, y1_pad, y2_pad),
+            mode='constant',
+            value=0)
+        # 1.4 pad attention mask
+        _, _, img_h, img_w = img_crop_padded.shape
+        att_mask = np.ones((img_h, img_w))
+        end_x, end_y = -x2_pad, -y2_pad
+        if y2_pad == 0:
+            end_y = None
+        if x2_pad == 0:
+            end_x = None
+        att_mask[y1_pad:end_y, x1_pad:end_x] = 0
+
+        # 2. Resize image and attention mask
+        resize_factor = output_size / crop_size
+        img_crop_padded_numpy = np.transpose(img_crop_padded.squeeze().cpu().numpy(),(1,2,0))
+        img_crop_padded_numpy = cv2.resize(img_crop_padded_numpy,(output_size, output_size))
+        img_crop_padded = torch.from_numpy(img_crop_padded_numpy).permute((2,0,1)).unsqueeze(0).to(img.device)
+
+        att_mask = cv2.resize(att_mask, (output_size, output_size)).astype(np.bool_)
+
+        return img_crop_padded, resize_factor, att_mask
+
+    def _normalize(self, img_tensor, amask_arr):
+        self.mean = torch.tensor([0.485, 0.456, 0.406]).view(
+            (1, 3, 1, 1)).cuda()
+        self.std = torch.tensor([0.229, 0.224, 0.225]).view(
+            (1, 3, 1, 1)).cuda()
```

Comment:
refer [here](https://github.com/open-mmlab/mmtracking/blob/master/mmtrack/core/motion/flow.py#L27) to change code